.github/workflows/regression-tests.yml

```diff
@@ -17,18 +17,26 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest]
-        compiler: [g++-10, g++-13]
+        compiler: [g++-10, g++-13, clang++-15]
         include:
           - os: ubuntu-20.04
             compiler: clang++-12
           - os: macos-13
             compiler: clang++
+          - os: macos-13
+            compiler: clang++-15
           - os: windows-latest
             compiler: cl.exe
     steps:
     - name: Checkout repo
       uses: actions/checkout@v4
 
+    - name: Prepare compilers
+      if: matrix.os == 'macos-13'
+      run: |
+        sudo xcode-select --switch /Applications/Xcode_14.3.1.app
```

Comment:
[Explanation] The path is taken from [here](https://github.com/actions/runner-images/blob/main/images/macos/macos-13-Readme.md).